llms/mlx_lm/server.py

```diff
@@ -80,244 +81,263 @@ def convert_chat(messages: any, role_mapping: Optional[dict] = None):
-    response = {
-        "id": chat_id,
-        "object": "chat.completion",
-        "created": int(time.time()),
-        "model": requested_model,
-        "system_fingerprint": f"fp_{uuid.uuid4()}",
-        "choices": [
-            {
-                "index": 0,
-                "message": {
-                    "role": "assistant",
-                    "content": text,
-                },
-                "logprobs": None,
-                "finish_reason": None,
-            }
-        ],
-        "usage": {
-            "prompt_tokens": len(prompt),
-            "completion_tokens": len(tokens),
-            "total_tokens": len(prompt) + len(tokens),
-        },
-    }
-
-    return response
-
-
-def create_completion_response(completion_id, requested_model, prompt, tokens, text):
-    return {
-        "id": completion_id,
-        "object": "text_completion",
-        "created": int(time.time()),
-        "model": requested_model,
-        "system_fingerprint": f"fp_{uuid.uuid4()}",
-        "choices": [
-            {"text": text, "index": 0, "logprobs": None, "finish_reason": "length"}
-        ],
-        "usage": {
-            "prompt_tokens": len(prompt),
-            "completion_tokens": len(tokens),
-            "total_tokens": len(prompt) + len(tokens),
-        },
-    }
-
-
-def create_chat_chunk_response(chat_id, requested_model, next_chunk):
-    response = {
-        "id": chat_id,
-        "object": "chat.completion.chunk",
-        "created": int(time.time()),
-        "model": requested_model,
-        "system_fingerprint": f"fp_{uuid.uuid4()}",
-        "choices": [
-            {
-                "index": 0,
-                "delta": {"role": "assistant", "content": next_chunk},
-                "logprobs": None,
-                "finish_reason": None,
-            }
-        ],
-    }
-    return response
-
-
-def create_completion_chunk_response(completion_id, requested_model, next_chunk):
-    return {
-        "id": completion_id,
-        "object": "text_completion",
-        "created": int(time.time()),
-        "choices": [
-            {"text": next_chunk, "index": 0, "logprobs": None, "finish_reason": None}
-        ],
-        "model": requested_model,
-        "system_fingerprint": f"fp_{uuid.uuid4()}",
-    }
-
-
 class APIHandler(BaseHTTPRequestHandler):
 
-    def __init__(self, *args, **kwargs):
-        # Prevent exposing local directory by deleting HEAD and GET methods
-        delattr(self, "do_HEAD")
-        delattr(self, "do_GET")
+    # System fingerprint is the same for all instances of the class
+    system_fingerprint = f"fp_{uuid.uuid4()}"
 
+    def __init__(self, *args, **kwargs):
+        """
+        Create static request specific metadata
+        """
+        self.created = int(time.time())
         super().__init__(*args, **kwargs)
 
-    def _set_headers(self, status_code=200):
+    def _set_completion_headers(self, status_code: int = 200):
```

Comment:
Since you are using it for both completion and 404 responses, maybe it would be better to just name it as _set_headers?

Reply:
I named it like this because it's clear that there are different headers for streams and completions. Maybe we just add a generic _add_header for cases where something goes wrong

Reply:
ok, in that case maybe just keep the imp as it is, don't  have to overcomplicate it.